Compilation Failure Due to Type Mismatch and Deprecated remove Method

My last bug was marked as closed, but I'm still unable to run a prover node on a fresh Ubuntu install, as of an hour ago. So here goes again:

🐛 Bug Report

I encountered a compilation failure while working with snarkOS. There are two primary issues noted:

    Deprecated usage of the remove method, which suggests using swap_remove instead to maintain map order.
    A type mismatch error (E0308), where a String was found when a Body was expected in an error handling function.

This bug prevents successful compilation of the snarkOS node and seems quite serious as it affects the fundamental operation of the software.
Steps to Reproduce

    Clone the snarkOS repository.
    Attempt to compile snarkOS using Rust.
    Observe the warnings about the deprecated remove method.
    Encounter the E0308 type mismatch error related to the .error_handler function expecting a Body type instead of a String.

Expected Behavior

The expected behavior is for snarkOS to compile without any errors or deprecation warnings. Instead of a String, the .error_handler should either be supplied with or be able to convert the provided string into a Body type, and the deprecated remove method should be updated to swap_remove.
Your Environment

    snarkOS Version: v2.2.7
    Rust Version: rustc 1.74.0 (79e9716c9 2023-11-13)
    Computer OS: Windows 10 Version 22H2 (OS Build 10945.3930)

I have the same issue. I tried using multiple different rustc version to solve the issue but it didn't work.
```

snarkOS Version: v2.2.7
Rust Version: rustc 1.76.0 (07dca489a 2024-02-04)
OS : ubuntu 22
```
```

Git branch - testnet3
Git log | head -n 1 - commit 937150eae125bd4d7c4b196d82acfe8f3390a035
```

Error while compiling -

```
Compiling snarkos-node-consensus v2.2.7 (/root/mit/snarkOS/node/consensus)
warning: `snarkos-node-bft` (lib) generated 5 warnings
   Compiling snarkos-node-rest v2.2.7 (/root/mit/snarkOS/node/rest)
error[E0308]: mismatched types
   --> node/rest/src/lib.rs:115:54
    |
115 |                 .error_handler(|error| Response::new(error.to_string()))
    |                                        ------------- ^^^^^^^^^^^^^^^^^ expected `Body`, found `String`
    |                                        |
    |                                        arguments to this function are incorrect
    |
note: associated function defined here
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/http-1.0.0/src/response.rs:253:12
    |
253 |     pub fn new(body: T) -> Response<T> {
    |            ^^^
help: call `Into::into` on this expression to convert `std::string::String` into `Body`
    |
115 |                 .error_handler(|error| Response::new(error.to_string().into()))
    |                                                                       +++++++

For more information about this error, try `rustc --explain E0308`.
error: could not compile `snarkos-node-rest` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

--- a/run-prover.sh
+++ b/run-prover.sh
@@ -39,7 +39,6 @@ while :
 do
   echo "Checking for updates..."
   git stash
-  rm Cargo.lock
   STATUS=$(git pull)
 
   if [ "$STATUS" != "Already up to date." ]; then

